### PR TITLE
Authorization Toggles

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Feature
[x] Refactoring (no functional changes, no API changes)

## What Is the New Behavior?

- retrieve user roles and permission for logged in user
- guard, directives and service for toggling storefront functionality depending on the current users permissions

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

## Other Information

- required ICM 7.10.20+ GA

### CODE REVIEW

File: src/app/pages/account/account-navigation/account-navigation.component.html [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
+          }}</a>
+        </li>
+      </ng-container>
       <ul *ngIf="item.value.children" class="account-navigation list-unstyled">
         <ng-container *ngFor="let subItem of item.value.children | keyvalue: unsorted">
-          <li>
+          <li *ishIsAuthorizedTo="item.value.permission || 'always'">
             <a
               *ishFeature="subItem.value.feature || 'always'"
               [routerLink]="item.key + subItem.key"
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/pages/account/account-navigation/account-navigation.component.html:21]: 👍

File: src/app/pages/account/account-navigation/account-navigation.component.html [hunk 2]
```diff
@@ -29,13 +32,15 @@
...
-        [value]="item.key"
-        [attr.selected]="item.value.link === currentPath ? 'selected' : undefined"
+      <ng-container *ishIsAuthorizedTo="item.value.permission || 'always'">
+        <option
+          *ishFeature="item.value.feature || 'always'"
+          [value]="item.key"
+          [attr.selected]="item.value.link === currentPath ? 'selected' : undefined"
+        >
+          {{ item.value.localizationKey | translate }}
+        </option></ng-container
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/pages/account/account-navigation/account-navigation.component.html:42]: crazy auto-formatting 😉

File: src/app/pages/account/account-navigation/account-navigation.component.ts [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
 
-import { AccountFacade } from 'ish-core/facades/account.facade';
 import { DeviceType } from 'ish-core/models/viewtype/viewtype.types';
 
 interface NavigationItems {
   [link: string]: {
     localizationKey: string;
     dataTestingId?: string;
     feature?: string;
-    children?: NavigationItems;
```
[the remark this answers is not in the document]
Comment (dhhyi) [reply src/app/pages/account/account-navigation/account-navigation.component.ts:14]: I was importing a revert from a previous commit and it seems I wasn't double checking...

**Comment by shauke** [comment]
The user info hover effect is removed - according to the changed IAD.

**Pull request merged**